src/data/fin/tuple/basic.lean

```diff
@@ -847,6 +847,34 @@ lemma mem_find_of_unique {p : fin n → Prop} [decidable_pred p]
 mem_find_iff.2 ⟨hi, λ j hj, le_of_eq $ h i j hi hj⟩
 
 end find
+section contract_nth
+
+variables {α : Type*}
+
+/-- Sends `(g₀, ..., gₙ)` to `(g₀, ..., op gⱼ gⱼ₊₁, ..., gₙ)`. -/
+def contract_nth (j : ℕ) (op : α → α → α) (g : fin (n + 1) → α) (k : fin n) : α :=
```

Comment:
```suggestion
def contract_nth (j : fin (n + 1)) (op : α → α → α) (g : fin (n + 1) → α) (k : fin n) : α :=
```
otherwise the semantics aren't clear when `j > n + 1

Reply:
(this also matches `insert_nth`)

Reply:
Ooh thank you, that makes sense, and it let me shorten a proof too.